src/Shared/Components/Security/Vulnerabilities/Vulnerabilities.tsx

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2024. Devtron Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useEffect } from 'react'
+import { EMPTY_STATE_STATUS, SCAN_TOOL_ID_TRIVY } from '@Shared/constants'
+import { SeverityCount } from '@Shared/types'
+import { Progressing, useAsync } from '../../../../Common'
+import { ScannedByToolModal } from '../../ScannedByToolModal'
+import { getLastExecutionByArtifactAppEnv } from './service'
+import { VulnerabilitiesProps } from './types'
+import { SecuritySummaryCard } from '../SecuritySummaryCard'
+import { getSecurityScan } from '../SecurityModal'
+
+const Vulnerabilities = ({
+    isScanned,
+    isScanEnabled,
+    artifactId,
+    applicationId,
+    environmentId,
+    setVulnerabilityCount,
+    isScanV2Enabled,
+}: VulnerabilitiesProps) => {
+    const [areVulnerabilitiesLoading, vulnerabilitiesResponse, vulnerabilitiesError, reloadVulnerabilities] = useAsync(
+        () => getLastExecutionByArtifactAppEnv(artifactId, applicationId, environmentId),
+        [],
+        isScanned && isScanEnabled && !isScanV2Enabled,
+        {
+            resetOnChange: false,
+        },
+    )
+
+    const [scanResultLoading, scanResultResponse, scanResultError, reloadScanResult] = useAsync(
+        () => getSecurityScan({ artifactId, appId: applicationId, envId: environmentId }),
+        [],
+        isScanned && isScanEnabled && isScanV2Enabled,
+        {
+            resetOnChange: false,
+        },
+    )
+
+    useEffect(() => {
+        if (scanResultResponse && isScanV2Enabled) {
+            setVulnerabilityCount(scanResultResponse?.result?.imageScan?.vulnerability?.list?.length)
+            return
+        }
+        if (vulnerabilitiesResponse && !isScanV2Enabled) {
+            setVulnerabilityCount(vulnerabilitiesResponse.result.vulnerabilities?.length)
+        }
+    }, [vulnerabilitiesResponse, scanResultResponse])
```

Comment:
Check if loading and error needs to be there?

Reply:
working fine